client/ayon_core/pipeline/farm/pyblish_functions.py

```diff
@@ -539,68 +546,60 @@ def _create_instances_for_aov(instance, skeleton, aov_filter, additional_data,
         ValueError:
 
     """
-    # TODO: this needs to be taking the task from context or instance
-    task = os.environ["AYON_TASK_NAME"]
+    task_name = instance.data["task"]
 
     anatomy = instance.context.data["anatomy"]
-    s_product_name = skeleton["productName"]
+    src_product_name = skeleton["productName"]
     cameras = instance.data.get("cameras", [])
-    exp_files = instance.data["expectedFiles"]
+    expected_files = instance.data["expectedFiles"]
     log = Logger.get_logger("farm_publishing")
 
     instances = []
     # go through AOVs in expected files
-    for aov, files in exp_files[0].items():
-        cols, rem = clique.assemble(files)
-        # we shouldn't have any reminders. And if we do, it should
-        # be just one item for single frame renders.
-        if not cols and rem:
-            if len(rem) != 1:
-                raise ValueError("Found multiple non related files "
-                                 "to render, don't know what to do "
-                                 "with them.")
-            col = rem[0]
-            ext = os.path.splitext(col)[1].lstrip(".")
-        else:
-            # but we really expect only one collection.
-            # Nothing else make sense.
-            if len(cols) != 1:
-                raise ValueError("Only one image sequence type is expected.")  # noqa: E501
-            ext = cols[0].tail.lstrip(".")
-            col = list(cols[0])
+    for aov, files in expected_files[0].items():
+        collected_files = _collect_expected_files_for_aov(files)
+
+        # get file path (use first from the list or single frame)
+        expected_filepath = collected_files[0] if isinstance(collected_files, (list, tuple)) else collected_files
+
+        dynamic_data = {
+            "aov": aov,
+            "renderlayer": instance.data.get("renderlayer"),
+        }
+
+        # find if camera is used in the file path
+        camera = [cam for cam in cameras if cam in expected_filepath]
+
+        # Is there just one camera matching?
+        # TODO: this is not true, we can have multiple cameras in the scene
+        #       and we should be able to detect them all. Currently, we are
+        #       keeping the old behavior, taking the first one found.
+        if camera:
+            dynamic_data["camera"] = camera[0]
+
+        product_name = get_product_name(
+            project_name=instance.context.data["projectName"],
+            task_name=task_name,
+            task_type=instance.data["taskEntity"]["taskType"],
+            host_name=instance.context.data["hostName"],
+            product_type=skeleton["productType"],
+            dynamic_data=dynamic_data,
+            variant=instance.data.get('variant', ''),
+            project_settings=instance.context.data.get("project_settings"),
+        )
```

Comment:
I'm a bit worried about the fact that this now seems to require templates to use `aov` to ensure multiple AOVs become a unique product? Does this mean that with 'older templates' this would now break? Or what is ensuring that they become separate products?